File: 202-HappyNumber/202-HappyNumber.c
// Last updated: 10/4/2025, 5:45:58 PM
bool isHappy(int n) {
    int sum, m;

    while (1) {
        sum = 0;

        // Get sum of squares of digits
        while (n != 0) {
            m = n % 10;
            sum += m * m;
            n = n / 10;
        }

        if (sum == 1)
            return true;
        else if (sum == 4)
            return false;  // 4 is part of the cycle for unhappy numbers

        n = sum;  // repeat the process
    }
}

File: 202-HappyNumber/202-HappyNumber.py
# Last updated: 10/4/2025, 5:45:56 PM
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        while True:
            total = 0
            while n != 0:
                digit = n % 10
                total += digit * digit
                n = n // 10

            if total == 1:
                return True
            elif total == 4:
                return False

            n = total
